src/robusta/core/sinks/slack/slack_sink_params.py

```diff
@@ -28,86 +10,6 @@ class SlackSinkParams(SinkBaseParams):
     channel_override: Optional[str] = None
     max_log_file_limit_kb: int = 1000
 
-    @validator("channel_override")
```

Comment:
hi @oscgu 
why are we changing the Slack parameters?

Reply:
Hi @arikalon1,
I moved out the `channel_override` into `src/robusta/core/sinks/common/channel_transformer.py`so that it can be reused accross sinks like so:

```python
 overriden_channel = ChannelTransformer.template(
                sink_params.channel_override,
                sink_params.slack_channel,
                self.cluster_name,
                finding.subject.labels,
                finding.subject.annotations,
            )
```
(Here in the slack sink: https://github.com/robusta-dev/robusta/pull/1363/files#diff-05a4c5c486a82e25806bd8351699c08cf038425841d6d55e9fdbd3b6f099b776R380)

Reply:
hey @oscgu 

I think all the parameters validators were deleted from the SlackSinkParams
the method `get_slack_channel` and some other methods

Reply:
I understand the reason for moving the `get_slack_channel` logic
but why do we need to remove the validation?

Reply:
@arikalon1 the validation of the `channel_override` param has also been moved to the `ChannelTransformer` , see [here](https://github.com/robusta-dev/robusta/pull/1363/files/1bc313fe73c5e32af5dc57f79ac5f0fc96133de6#diff-9b26b848df0cca0c340d005d0ea7b912f92fa00b2911045341b54437c047f7abR78).
Should that rather be done in the sinks themselves, with the validator annotation?

Reply:
I guess it should happen in the sink or else it will get re-validated with each invocation of the template func, even though it never changes

Reply:
yeah, I agree

I think the validation should happen only once, when creating the params object

This is also create after a configuration change, and an error at this phase cause the pod to crash, and the user should notice the configuration error